src/extract_qe.py

```diff
@@ -189,42 +189,59 @@ def qe_extract_gradients(inp_str,  flag):
 
 #alp_d,bet_d = check_eig_deg()
 
-def fermi_pop(e,nel,nspin,kT):
+def fermi_pop(e,nel,nspin,kT,el_st):
     ##
     # This function generates occupation scheme based on fermi population
     # \param[in] e   List of eigen energies of the Molecular orbitals
     # \param[in] nel Total number of electrons in the system
     # \params[out] occ_new  List of fermi population of the MOs
     # \params[in] nspin Spin, = 2,1 for non-polarized and spin-polarized calculations.
+    # \params[in] el_st electronic state index; different fractional scheme for different electronic states.
 
     N = len(e)  # Total number of MOs in the active space
-    a = MATRIX(N,N)
-    for i in xrange(N):
-        for j in xrange(N):
-            if i==j:
-                a.set(i,j, e[i])
-            else:
-                a.set(i,j, 0.0)
-    if nspin == 2:  # For spin-polarized calculations.
-        Nel = nel/2 + nel%2  # Number of electrons in the alpha or beta spin orbital
-        degen = 1.0 # One orbital can have 1 electrons, in this case of spin-polarization
-    if nspin == 1:  # For non-polarized calculations
-        Nel = nel  # Total number of electrons.
-        degen = 2.0 # One orbital can have 2 electrons, in case of non-polarized calculations
-    #kT = 0.000955482 # kT = 0.000955482 Hartree, = 0.02585 eV  # at 300K
+    pop_opt,occ_tot,occ_new = 1,[],[]
     etol = 0.0000000001
-    #Ef = fermi_energy(e, Nel, degen, kT, etol)  # Not needed
-    bnds = order_bands(a)
-    #print bnds
-    #print "\n Test5: populate bands"
-    pop_opt = 1
-    pop_fermi = populate_bands(Nel, degen, kT, etol, pop_opt, bnds)
+    a = MATRIX(N,N)
+    if el_st==0: # For S0 
+        el_scheme = [0]
+    elif el_st==1: # For S1
+        el_scheme = [-1,0,1]
+    elif el_st==2:  # For S2
```

Comment:
please, add the condition (e.g. Warning and exit) if el_st is anything other than 0, 1, or 2